src/NServiceBus.RabbitMQ.Tests/RabbitMqContext.cs

```diff
@@ -99,30 +69,39 @@ public void TearDown()
             channelProvider?.Dispose();
         }
 
-        protected IncomingMessage WaitForMessage()
+        protected bool TryWaitForMessageReceipt()
         {
-            var waitTime = TimeSpan.FromSeconds(1);
+            IncomingMessage message;
+            return TryReceiveMessage(out message, incomingMessageTimeout);
+        }
 
-            if (Debugger.IsAttached)
+        protected IncomingMessage ReceiveMessage()
+        {
+            IncomingMessage message;
+            if (!TryReceiveMessage(out message, incomingMessageTimeout))
             {
-                waitTime = TimeSpan.FromMinutes(10);
+                throw new TimeoutException($"The message did not arrive within {incomingMessageTimeout.TotalSeconds} seconds.");
             }
 
-            IncomingMessage message;
-            receivedMessages.TryTake(out message, waitTime);
-
             return message;
         }
 
+        private bool TryReceiveMessage(out IncomingMessage message, TimeSpan timeout) =>
+            receivedMessages.TryTake(out message, incomingMessageTimeout);
+
+        protected virtual IEnumerable<string> AdditionalReceiverQueues => Enumerable.Empty<string>();
+
         protected const string ReceiverQueue = "testreceiver";
         protected const string ErrorQueue = "error";
         protected MessageDispatcher messageDispatcher;
         protected ConnectionFactory connectionFactory;
-        private ChannelProvider channelProvider;
         protected MessagePump messagePump;
-        BlockingCollection<IncomingMessage> receivedMessages;
-
-        protected ConventionalRoutingTopology routingTopology;
         protected SubscriptionManager subscriptionManager;
+
+        private ChannelProvider channelProvider;
+        private BlockingCollection<IncomingMessage> receivedMessages;
```

Comment:
Given that BlockingCollection blocks, I wonder if we should replace this with something non-blocking now that we have async stuff everywhere. For example, we could use `BufferBlock<T>` from the [TPL Dataflow](https://www.nuget.org/packages/Microsoft.Tpl.Dataflow/) library instead.